src/ui/pages/slack_alerts_setup/form.jsx

```diff
@@ -12,24 +12,74 @@ import { ButtonGroup, FormWrapper } from './styles';
 export function SetupForm() {
     const [{ loading, error, data, isDirty }, update, save] = useSlackConfig();
 
+    const updateOauthToken = useCallback((e, { value }) => update({ ...data, slack_app_oauth_token: value }));
     const updateUrl = useCallback((e, { value }) => update({ ...data, webhook_url: value }));
     const updateUser = useCallback((e, { value }) => update({ ...data, from_user: value }));
     const updateUserIcon = useCallback((e, { value }) => update({ ...data, from_user_icon: value }));
 
+    const slackAppOauthToken = loading ? '' : data.slack_app_oauth_token;
```

Comment:
```suggestion
    const slackAppOauthToken = loading ? '' : data.slack_app_oauth_token;
```

Lack of Splunk knowledge, but does Splunk have somewhere to store secrets as I would assume that would be the more preferable approach than allowing configuration within each alert action and having that visible to anyone with the appropriate permissions?

Reply:
Hey @swelcher - Sorry for the delay here.

Splunk does have a configuration file for storing default OAuth tokens. That config file is `alert_actions.conf`. You can see an example of this config file in `alert_actions.conf.spec` in this change set.

In this case, the Slack alert action allowed you to provide that default token in the conf file, but it also let users override that token for individual alerts. Those override tokens would be visible to anyone on the front end. I biased towards preserving that override functionality.

Let me know if you think we should get rid of that override option. I'd be happy to do so (assuming @ziegfried is also OK with removing that functionality).

Reply:
I thought about this too. I think it would be ideal if the user could configure multiple Slack workspaces and possibly even channels in app setup and then select one when creating an alert. It would be quite a bit of work but should be possible. 

It would be possible to store secrets in the [app secret storage](https://dev.splunk.com/enterprise/docs/developapps/manageknowledge/secretstorage), but permissions can get a bit tricky and I opted to keep it simple. Possibly something to look into in the future.